Src/Fido2/Extensions/CryptoUtils.cs

```diff
@@ -59,7 +59,9 @@ public static bool ValidateTrustChain(X509Certificate2[] trustPath, X509Certific
         // A trust anchor can be a root certificate, an intermediate CA certificate or even the attestation certificate itself.
 
         // Let's check the simplest case first.  If subject and issuer are the same, and the attestation cert is in the list, that's all the validation we need
-        if (trustPath.Length == 1 && trustPath[0].Subject.Equals(trustPath[0].Issuer, StringComparison.Ordinal))
+
+        // We have the same singular root cert in trustpath and it is in attestationRootCertificates
+        if (trustPath.Length == 1)
```

Comment:
@aseigler Is it really ok to *not* equality compare the path?

Without this we get: 
![CleanShot 2024-07-16 at 11 23 02](https://github.com/user-attachments/assets/550a0bfe-6fad-40ad-9f09-2b02402cb246)

I could throw in a `|| conformanceTesting`-related check

Reply:
The check to see if the subject and the issuer are the same is based on logic described in https://datatracker.ietf.org/doc/html/rfc5280#section-3.2.  I thought we had already covered this before, where the batch certificate referenced in the metadata is not a root...I'll have to review this more thoroughly.

Reply:
@abergs I saw that we removed the check to make sure the trustPath's root subject is equal to the issuer. 

What does the conformance test use for the Subject and Issuer (where they don't match). Is there a case, in production, where these values would also not match? Are they completely different values, or just encoded differently?

Reply:
@iamcarbon Not sure - I haven't looked into it.

Reply:
As part of #554 I reverted this change until we get more data on this or @aseigler can review. I will open another PR to bring these changes back.